provisioning/transport/http/src/ProvisioningTransportHandlerHttp.cs

```diff
@@ -148,10 +148,34 @@ public ProvisioningTransportHandlerHttp()
 
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    operation = await client.RuntimeRegistration.OperationStatusLookupAsync(
+                    try
+                    {
+                        operation = await client.RuntimeRegistration.OperationStatusLookupAsync(
                         registrationId,
                         operationId,
                         message.IdScope).ConfigureAwait(false);
+                    }
+                    catch(HttpOperationException oe) when (oe.Response.StatusCode >= (HttpStatusCode)429)
```

Comment:
this also applies to all 500 error codes ... should we be catching 429 specifically ?

Reply:
Also if we decide to catch 500s, the header won't have the retry-after header as it will not be Too Many Request. In that case, we need to set it to default value.

Reply:
That's a good question. 

@vinagesh There is a logic above where retry-after is null it will set it to default value of 3s but the major question here is in case of internal error or something do we want to retry or not. I assume it to be safe to retry only for 429, in case of any other error just throw exception.

Tagging @prmathur-microsoft to confirm if it is safe to assume the above.

Reply:
I don't think this logic as it is works in our favor, we have to be a lot more specific if we want to catch a range of exception codes. Best would be to only catch 429s and let the natural flow go through otherwise.

Reply:
I believe I do have an answer. This catch block logic would be for any transient errors. I have made necessary changes, re-review new commit.